File: cpp/two-sum.cpp
/* Problem url: https://leetcode.com/problems/two-sum
 * Code by: ldcduc
 * */
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); ++ i) {
            if (m.count(target - nums[i]) > 0) {
                result.push_back(m[target - nums[i]]);
                result.push_back(i);
            }
            m[nums[i]] = i;
        }
        return result;
    }
};
/*
 * Comment by ldcduc
 * Suggested tags: hash map
 * 
 * */

File: go/two-sum.go
func twoSum(nums []int, target int) []int {
    var m map[int]int = make(map[int]int)    
    for i, value := range nums {
        if m[target - value] != 0{ 
            return []int {m[target - value] - 1, i}
        } 
        m[value] = i + 1
    }
    return []int {}
}
